src/customGraphQLValidationRules.js

```diff
@@ -12,7 +12,12 @@ export function OperationsMustHaveNames(context) {
   };
 }
 
-function getFieldWasRequestedOnNode(node, field) {
+function mustHaveSelection(type) {
+  if(!type) return false
+  return isCompositeType(type) || mustHaveSelection(type.ofType)
+}
+
+function fieldIsRequestedOnNode(node, field) {
```

Comment:
I have changed this function name to make it more consistent with other functions (`fieldAvailableOnType` for example). Since it's private, it should not have any impact outside of this file.

Same thing here, if you don't want refactoring, I can revert and submit only the bug fix.